onnx/defs/quantization/defs.cc

```diff
@@ -199,10 +223,17 @@ ONNX_OPERATOR_SET_SCHEMA(
         .TypeConstraint(
             "T2",
             {"tensor(float)", "tensor(float16)", "tensor(bfloat16)"},
-            "'x_scale' determines the output type.")
-        .SetDoc(DequantizeLinear_ver21_doc)
+            "The type of the input 'x_scale'.")
+        .TypeConstraint("T3", {"tensor(float)", "tensor(float16)", "tensor(bfloat16)"}, "The type of the output 'y'.")
+        .SetDoc(DequantizeLinear_ver23_doc)
         .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
-          propagateElemTypeFromInputToOutput(ctx, 1, 0);
+          auto const output_dtype =
+              static_cast<TensorProto_DataType>(getAttribute(ctx, "output_dtype", TensorProto::UNDEFINED));
+          if (output_dtype != TensorProto::UNDEFINED) {
```

Comment:
I wonder if we should use `ctx.getAttribute("output_dtype") != nullptr` instead? Just wonder what should happen in the edge-case if the user specifies this attribute with a value of UNDEFINED==0? Should we treat it as an error, or as if the attribute was not specified? 

This may be relevant to the reference-implementation ... if 0 is legal, we should make the reference implementation support it.

Reply:
Looking at the existing behavior of `output_dtype` in `QuantizeLinear`, I think 0 should be legal and it should be treated as if no value is provided.

It doesn't look like the numpy ref implementation of `output_dtype` in `QuantizeLinear` uses TensorProto though. Is this a mistake?

Reply:
I made the newly added attributes take default values of `static_cast<int64_t>(0)`, ie `TensorProto.UNDEFINED`, in keeping with the existing attributes. This way this check should be sufficient to cover both the 0 and null cases.

I also updated the python ref implementations.